File: 1647_Minimum Deletions to Make Character Frequencies Unique/MinimumDeletionstoMakeCharacterFrequenciesUnique.py
class Solution:
    def minDeletions(self, s: str) -> int:
        count, ans, used = collections.Counter(s), 0, set()
        for c, freq in count.items():
            while freq > 0 and freq in used:
                freq -= 1
                ans += 1
            used.add(freq)
        return ans

File: 1647_Minimum Deletions to Make Character Frequencies Unique/MinimumDeletionstoMakeCharacterFrequenciesUnique.java
class Solution {
    public int minDeletions(String s) {
        int[] count = new int[26];
        for(char c : s.toCharArray())
            count[c - 'a']++;
        
        var ans = 0;
        HashSet<Integer> set = new HashSet();
        
        for(int i=0;i<count.length;i++)
        {
            while(count[i] > 0 && !set.add(count[i]))
            {
                count[i]--;
                ans++;
            }
        }
        
        return ans;
    }
}